src/shared/dbpath.cc

```diff
@@ -19,18 +19,65 @@
  */
 
 #include "dbpath.h"
+#include "log.h"
 
 #include <stdlib.h>
+#include <stdio.h>
+#include <fstream>
+#include <string.h>
+#include <stdlib.h>
+#include "filesystem.h"
 
 std::string url = std::string("mysql:db=box_utf8;user=") +
```

Comment:
While we are here, perhaps refactor to use a single string-generation routine (or macro) to initialize and re-set the `url` with same code? Copies like these need extra revision to make sure they do the same thing...